.github/workflows/check-certificates.yml

```diff
@@ -0,0 +1,99 @@
+name: Check for issues with signing certificates
+
+on:
+  schedule:
+    # run every 10 hours
+    - cron: "0 */10 * * *"
+  # workflow_dispatch event allows the workflow to be triggered manually.
+  # This could be used to run an immediate check after updating certificate secrets.
+  # See: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_dispatch
+  workflow_dispatch:
+
+env:
+  # Begin notifications when there are less than this many days remaining before expiration
+  EXPIRATION_WARNING_PERIOD: 30
+
+jobs:
+  get-certificates-list:
+    # This workflow would fail in forks that don't have the certificate secrets defined
+    if: github.repository == 'arduino/FirmwareUpdater'
+    runs-on: ubuntu-latest
+    outputs:
+      certificates: ${{ steps.get-certificates.outputs.certificates }}
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: Set certificates path environment variable
+        run: |
+          # See: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
+          echo "FILES=\\\"$(ls -m ${{ github.workspace }} | sed 's/, /\\",\\"/')\\\"" >> $GITHUB_ENV
+
+      - name: Get files list
+        id: get-files
+        run: |
+          JSON=$(echo "[${{ join(env.FILES) }}]" | jq -c '[{"cert_file": .[]}]')
+          echo "::set-output name=certificates::{\"include\": $JSON }"
```

Comment:
I prefered going this way just in case when adding new certificates in the future we forget to add them to the list in the workflows. But thinking about it that's not probably gonna happen, I'll change it as you suggest.

Reply:
It's quite possible that I have not understood the reason for using `include`. As I said, I was quite confused and had to run some experiments to even understand if it would work. But I wasn't able to fully use the system you set up due to the `sed` quoting problem I reported above.

I've only ever used that feature before for things like this:
https://github.com/SpenceKonde/megaTinyCore/blob/78d086fe3f48742cffe21ba63317bf023bf2e76c/.github/workflows/compile-examples.yml#L135
If there are benefits to the current approach, I have no problems with it. From my quick experiements, it seems to work fine.

Reply:
I mistakenly followed this example: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-returning-a-json-object
Thought that you had use `include`. :sweat:

Reply:
>I prefered going this way just in case when adding new certificates in the future we forget to add them to the list in the workflows. But thinking about it that's not probably gonna happen, I'll change it as you suggest.

Just to make sure there isn't any misunderstanding. I'm not at all advocating hardcoding the list of certificates. I think your approach of generating the matrix from the directory listing is the best way. The example I showed above:
```yaml
strategy:
  matrix:
    cert_file:
      - certs/AddTrustExternalCA_Root.cer
      - certs/BaltimoreCyberTrust_Root.cer
...
```
was only intended to show the structure of the data that would be generated by the the code in the `get-certificates-list`.

Reply:
Aaaah ok, I'll go with the matrix generation then. :woman_technologist:

Reply:
Sorry for the confusion. I didn't do a good job at communicating. I was intending to do another pass though the review comments before submitting it, but then saw action on the PR just when I was getting toward the end of the review so I figured I'd better just submit it right away.

The generated matrix is super awesome.

Reply:
Don't worry, it's fine. :)